libs/platform/ebpf_crypto_hash.c

```diff
@@ -68,8 +68,11 @@ _Must_inspect_result_ ebpf_result_t
 ebpf_cryptographic_hash_append(
     _Inout_ ebpf_cryptographic_hash_t* hash, _In_reads_bytes_(length) const uint8_t* buffer, size_t length)
 {
-    NTSTATUS nt_status;
+    if (buffer == NULL) {
+        return EBPF_INVALID_ARGUMENT;
```

Comment:
Since the SAL annotation is `_In_` not `_In_opt_` then the only way for this to happen is for the SAL annotation to be violated by the caller.  As such, this should not gracefully fail if `_In_` is kept, it should indeed crash if NULL is passed.  Instead either change the SAL annotation or change the caller to not pass null.

Reply:
The function signature In_reads_bytes_(length) const uint8_t* buffer specifies that the function takes a pointer to a buffer of length bytes that are read-only and are not modified by the function. Therefore, it is possible for the buffer to be null if the length is zero.

Reply:
`_In_reads_bytes_` is a SAL annotation that means the buffer must not be null.
If it can be null then the annotation should be `_In_reads_bytes_opt_`.
See https://learn.microsoft.com/en-us/cpp/code-quality/annotating-function-parameters-and-return-values?view=msvc-170 for docs.

Reply:
Good point.  Also, in the latest, I'm unable to reproduce the test case to fail. Hence will revert the code.